lms/djangoapps/support/migrations/0003_course_reset.py

```diff
@@ -0,0 +1,49 @@
+# Generated by Django 3.2.23 on 2024-02-22 14:48
+
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
+import django.utils.timezone
+import model_utils.fields
+import opaque_keys.edx.django.models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ('student', '0045_auto_20230808_0944'),
+        ('support', '0002_alter_historicalusersocialauth_options'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='CourseResetCourseOptIn',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('created', model_utils.fields.AutoCreatedField(default=django.utils.timezone.now, editable=False, verbose_name='created')),
+                ('modified', model_utils.fields.AutoLastModifiedField(default=django.utils.timezone.now, editable=False, verbose_name='modified')),
+                ('course_id', opaque_keys.edx.django.models.CourseKeyField(max_length=255)),
+                ('active', models.BooleanField()),
+            ],
+            options={
+                'abstract': False,
+            },
+        ),
```

Comment:
Is there a reason to make this a model instead of using a Waffle Flag? I feel like it gives us *less* capability than a Waffle flag (e.g. no org override), at greater development cost (having to create a new model).

Reply:
We use many waffle flags for many different things. My thought was that having this be its own thing would be best to minimize the chance of getting mixed up with other flags. Also, other flags feel more .. "behind the scenes" to me? If that makes sense? I'm not thinking of this as functionality we're turning off / on but rather a feature that we are configuring via its own admin interface.